src/server/pfs/server/driver.go

```diff
@@ -2695,6 +2793,9 @@ func (d *driver) createBranch(txnCtx *txnenv.TransactionContext, branch *pfs.Bra
 			}
 			add(&branchInfo.DirectProvenance, provBranch)
 		}
+		if trigger != nil {
```

Comment:
Lol, no good point.

Reply:
Looked at this a little more and realized this `if` actually is doing something. It makes it so you can do `pachctl create branch repo@branch --head commit` on a branch that has a trigger without deleting the trigger. Which is a nice workflow to support. It also means there's no way to clear the trigger on a branch without deleting and recreating the branch. I think clearing triggers is probably a pretty uncommon workflow compared to changing the head of the branch and I believe we do a similar thing with provenance... so I think these semantics make sense. They do seem a little weird to me though, so I'm not completely sure.

Reply:
Actually this is a little bit broken for sure because the way `create branch` works it sends a non-nil trigger with an empty branch when it's unset, so the api should account for that.